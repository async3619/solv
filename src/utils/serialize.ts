import * as util from "util";

export function serialize(item: any, json = false) {
    const serializer: (target: any) => string = json ? JSON.stringify : util.inspect;
    return typeof item === "string" ? item : serializer(item);
}
