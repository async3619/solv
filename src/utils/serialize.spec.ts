import * as util from "util";
import { serialize } from "./serialize";

describe("serialize", () => {
    it("should serializes string data", () => {
        expect(serialize("Test")).toBe("Test");
    });

    it("should serializes non-string data", () => {
        jest.mock("util");

        let lastData: any = null;
        (util.inspect as any) = (item: any) => (lastData = item);

        serialize(123);
        expect(lastData).toBe(123);
    });

    it("should serializes data with json flag set", () => {
        expect(serialize({}, true)).toBe("{}");
        expect(serialize(123, true)).toStrictEqual("123");
        expect(serialize("123", true)).toMatchSnapshot();
    });
});
