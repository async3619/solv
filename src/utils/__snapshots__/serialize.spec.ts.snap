// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serialize should serializes data with json flag set 1`] = `"123"`;
